pkg/didcomm/packager/packager.go

```diff
@@ -195,22 +185,5 @@ func (bp *Packager) UnpackMessage(encMessage []byte) (*transport.Envelope, error
 		return nil, fmt.Errorf("unpack: %w", err)
 	}
 
-	//	ignore error - agents can communicate without using DIDs - for example, in DIDExchange
-	theirDID, err := bp.connectionStore.GetDID(base58.Encode(envelope.FromKey))
-	if errors.Is(err, did.ErrNotFound) {
-	} else if err != nil {
-		return nil, fmt.Errorf("failed to get their did: %w", err)
-	}
-
-	// ignore error - at beginning of DIDExchange, you might be about to generate a DID
-	myDID, err := bp.connectionStore.GetDID(base58.Encode(envelope.ToKey))
-	if errors.Is(err, did.ErrNotFound) {
-	} else if err != nil {
-		return nil, fmt.Errorf("failed to get my did: %w", err)
-	}
-
-	envelope.ToDID = myDID
-	envelope.FromDID = theirDID
```

Comment:
I am confused by the removal of both `myDID` and `theirDID`.

Are we losing information here or is it being supplied somewhere else? How are users of the framework affected by this removal?

Reply:
these properties were used just to carry information from [here](https://github.com/hyperledger/aries-framework-go/blob/d24bb59ffab793f895032ad9356251bb1b2092a3/pkg/didcomm/packager/packager.go#L212) to [here](https://github.com/hyperledger/aries-framework-go/blob/d24bb59ffab793f895032ad9356251bb1b2092a3/pkg/didcomm/transport/ws/pool.go#L111), it was updated as retrieving DIDs should not be the responsibility of `packager.UnpackMessage()`

Reply:
How are users of the framework affected by this removal?

Reply:
* in the framework these properties were used only in the transport layer (http and ws) and introduced when support for inbound/outbound messaging was added ([commit](https://github.com/hyperledger/aries-framework-go/commit/5d321d4181ffb8eedf4ddb43364ee1c75fea28dc)) - it looks more like some sort of workaround or quick "solution" - logic for DID retrieval is moving now to the message handler - and it will act as an orchestrator (reasonable role for it)
* I think that using properties of `transport.Envelope` outside of the framework should be considered as an error (this is low-level details of our implementation rather than declared contract from our side)

Reply:
@aholovko 

> these properties were used just to carry information from [here](https://github.com/hyperledger/aries-framework-go/blob/d24bb59ffab793f895032ad9356251bb1b2092a3/pkg/didcomm/packager/packager.go#L212) to [here](https://github.com/hyperledger/aries-framework-go/blob/d24bb59ffab793f895032ad9356251bb1b2092a3/pkg/didcomm/transport/ws/pool.go#L111), it was updated as retrieving DIDs should not be the responsibility of `packager.UnpackMessage()`

`myDID` and `theirDID` travel further than the code you pointed to: https://github.com/hyperledger/aries-framework-go/blob/d24bb59ffab793f895032ad9356251bb1b2092a3/pkg/didcomm/transport/ws/pool.go#L111

The end up being included in `DIDCommAction` events, eg. [`introduce` protocol](https://github.com/hyperledger/aries-framework-go/blob/f1c7f7aa39eca010c0c14cdb8ea7e0287db52368/pkg/didcomm/protocol/introduce/properties.go#L25-L32), making this a breaking change to the API.

Reply:
Here is the issuer adapter using these two properties during the issuance flow:

https://github.com/trustbloc/edge-adapter/blob/2835b01720871ae3a74d88ab4247aafd6d14d26e/pkg/restapi/issuer/operation/operations.go#L1362-L1379

Reply:
@llorllale Introduce protocol initializes both `myDID` and `theirDID` properties of `metaData` structure using input parameters of [HandleInbound](https://github.com/hyperledger/aries-framework-go/blob/6c1b27989cc2d17adaac330504b2ddcb440d913e/pkg/didcomm/protocol/introduce/service.go#L296) method. We still pass those data to the `HandleInbound` method - it's just fetched now in [InboundMessageHandler](https://github.com/hyperledger/aries-framework-go/blob/6c1b27989cc2d17adaac330504b2ddcb440d913e/pkg/framework/context/context.go#L183) and not in `UnpackMessage` of packager.

Reply:
Fair.